components/dashboard/src/projects/ProjectSettings.tsx

```diff
@@ -90,16 +83,73 @@ export default function () {
                 checked={!project.settings?.keepOutdatedPrebuildsRunning}
                 onChange={({ target }) => updateProjectSettings({ keepOutdatedPrebuildsRunning: !target.checked })}
             />
+            <CheckBox
+                title={
+                    <span>
+                        Use Last Successful Prebuild{" "}
+                        <PillLabel type="warn" className="font-semibold mt-2 ml-2 py-0.5 px-2 self-center">
+                            Alpha
+                        </PillLabel>
+                    </span>
+                }
+                desc={
+                    <span>
+                        Skip waiting for prebuilds in progress and use the last successful prebuild from previous
+                        commits on the same branch.
+                    </span>
+                }
+                checked={!!project.settings?.allowUsingPreviousPrebuilds}
+                onChange={({ target }) =>
+                    updateProjectSettings({
+                        allowUsingPreviousPrebuilds: target.checked,
+                        // we are disabling prebuild cancellation when incremental workspaces are enabled
+                        keepOutdatedPrebuildsRunning: target.checked || project?.settings?.keepOutdatedPrebuildsRunning,
+                    })
+                }
+            />
+            <div className="flex mt-4 max-w-2xl">
+                <div className="flex flex-col ml-6">
+                    <label
+                        htmlFor="prebuildNthCommit"
+                        className="text-gray-800 dark:text-gray-100 text-md font-semibold cursor-pointer tracking-wide"
+                    >
+                        Skip Prebuilds
+                    </label>
+                    <input
+                        type="number"
+                        id="prebuildNthCommit"
+                        min="0"
+                        max="100"
+                        step="5"
+                        className="mt-2"
+                        disabled={!project.settings?.allowUsingPreviousPrebuilds}
+                        value={
+                            project.settings?.prebuildEveryNthCommit === undefined
+                                ? 0
+                                : project.settings?.prebuildEveryNthCommit
+                        }
+                        onChange={({ target }) =>
+                            updateProjectSettings({
+                                prebuildEveryNthCommit: Math.abs(Math.min(Number.parseInt(target.value), 100)) || 0,
+                            })
+                        }
+                    />
+                    <div className="text-gray-500 dark:text-gray-400 text-sm mt-2">
+                        The number of commits that are skipped between prebuilds.
+                    </div>
```

Comment:
We'll likely iterate on this feature and the UX more later, but I wonder: Why bundle these two features together? (I.e. "run prebuilds every N commits" and "always skip prebuild in progress").

I feel like these should be two separate features, and I'm not 100% sure we can do a good job explaining to users exactly what this "number of skipped commits" does and how to use it properly for their project.